src/docfx/build/toc/parser/MarkdownTocMarkup.cs

```diff
@@ -37,9 +37,9 @@ public static (List<Error> errors, TableOfContentsInputModel model) LoadMdTocMod
 
             var (metaErrors, metadata) = ExtractYamlHeader.Extract(file, context);
             errors.AddRange(metaErrors);
-            var tocModel = new TableOfContentsInputModel
+            var tocModel = new TableOfContentsModel
             {
-                Metadata = metadata,
+                Metadata = metadata.ToObject<TableOfContentsMetadata>(),
```

Comment:
validate TOC YAML header here?

Reply:
Second thought: BuildPage needs to warn reserved metadata because we generate these additional metadata in output, but for TOC, we didn't generate these metadata. From that perspective, we may not warn here.